server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationAllocationIT.java

```diff
@@ -54,19 +54,50 @@ public void enablePreferPrimaryBalance() {
             client().admin()
                 .cluster()
                 .prepareUpdateSettings()
-                .setPersistentSettings(
-                    Settings.builder().put(BalancedShardsAllocator.PREFER_PER_INDEX_PRIMARY_SHARD_BALANCE.getKey(), "true")
-                )
+                .setPersistentSettings(Settings.builder().put(BalancedShardsAllocator.PREFER_PRIMARY_SHARD_BALANCE.getKey(), "true"))
         );
     }
 
+    /**
+     * This test verifies that the overall primary balance is attained during allocation. This test verifies primary
+     * balance per index and across all indices is maintained.
+     * @throws Exception
+     */
+    public void testGlobalPrimaryAllocation() throws Exception {
```

Comment:
Should we also add a test case here for allocation/rebalance post failover with these constraints?  To simulate rebalance when a shard comes back full of replicas.

Reply:
Re-balancing is only performed at per index level today. So, allocation on a multiple index failover scenario wouldn't be predictible. The per index allocation/rebalancing scenario is covered in existing ITs.